vendor/github.com/hashicorp/raft/net_transport.go

```diff
@@ -151,10 +154,26 @@ func NewNetworkTransportWithLogger(
 		timeout:      timeout,
 		TimeoutScale: DefaultTimeoutScale,
 	}
-	go trans.listen()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	trans.streamCancel = cancel
+	go trans.listen(ctx)
 	return trans
 }
 
+// Pause closes the current stream for a NetworkTransport instance
+func (n *NetworkTransport) Pause() {
```

Comment:
I would prefer these to be called `CloseStream` and `UseStream` and document that calling `CloseStream` must be done before giving a new stream layer